tests/ops/qubit/test_parametric_ops.py

```diff
@@ -2780,13 +2783,17 @@ def test_PauliRot_all_Identity_broadcasted(self):
         op = qml.PauliRot(theta, "II", wires=[0, 1])
         decomp_ops = op.decomposition()
 
-        phases = np.exp(-1j * theta / 2)
-        assert np.allclose(op.eigvals(), np.outer(phases, np.ones(4)))
-        mat = op.matrix()
-        for phase, sub_mat in zip(phases, mat):
-            assert np.allclose(sub_mat, phase * np.eye(4))
+        assert len(decomp_ops) == 1
+
+        decomp_op = decomp_ops[0]
+
+        assert qml.equal(decomp_op, qml.GlobalPhase(theta / 2))
 
-        assert len(decomp_ops) == 0
+        op_matrices = op.matrix()
+        decomp_op_matrices = decomp_op.matrix().T
```

Comment:
```suggestion
        decomp_op_matrices = qml.math.tensordot(decomp_op.matrix(), np.eye(4), axes=0)
        assert qml.math.allclose(op_matrices, decomp_op_matrices)
```

Reply:
I think this is a good idea in source code where it might provide a small edge on performance, but here it makes it less readable which I think should be a priority when writing tests. Also the performance improvement is not significant due to the scales we work with here